File: binary-tree-tilt/binary-tree-tilt.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def findTilt(self, root: TreeNode) -> int:
        '''DFS Approach to Find Tilt'''
        if not root: return 0
        tilt_sum = 0
        tilt_sum = abs(self.dfs(root.left) - self.dfs(root.right))
        return tilt_sum + self.findTilt(root.left) + self.findTilt(root.right)
    
    def dfs(self, root: TreeNode) -> int:
        if not root: return 0
        return root.val + self.dfs(root.left) + self.dfs(root.right)

File: binary-tree-tilt/binary-tree-tilt.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution 
{
    public int findTilt(TreeNode root) 
    {
        if (root == null) return 0;
        
        // Calculate the tilt of the current node
        int localTilt = 0;
        localTilt = Math.abs(sumDFS(root.left) - sumDFS(root.right));
        
        // Calculate the sum of the tilts of all nodes
        return localTilt + findTilt(root.left) + findTilt(root.right);
    }
    
    // Calculates the sum of any given subtree
    private int sumDFS(TreeNode root)
    {
        if (root == null) return 0;
        return root.val + sumDFS(root.left) + sumDFS(root.right);
    }
}